### ISSUE TITLE
ZeroDivisionError: float division by zero

### ISSUE THREAD

**Issue by geoyogesh** [DrewThomasson/ebook2audiobook#47]
Error while running the latest docker 
```
 > Processing time: 0.0005831718444824219
Chapter 0:   2%|▌                             | 9/437 [11:08<8:49:48, 74.27s/it]
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/site-packages/gradio/queueing.py", line 536, in process_events
    response = await route_utils.call_process_api(
  File "/usr/local/lib/python3.10/site-packages/gradio/route_utils.py", line 322, in call_process_api
    output = await app.get_blocks().process_api(
  File "/usr/local/lib/python3.10/site-packages/gradio/blocks.py", line 1935, in process_api
    result = await self.call_function(
  File "/usr/local/lib/python3.10/site-packages/gradio/blocks.py", line 1520, in call_function
    prediction = await anyio.to_thread.run_sync(  # type: ignore
  File "/usr/local/lib/python3.10/site-packages/anyio/to_thread.py", line 56, in run_sync
    return await get_async_backend().run_sync_in_worker_thread(
  File "/usr/local/lib/python3.10/site-packages/anyio/_backends/_asyncio.py", line 2405, in run_sync_in_worker_thread
    return await future
  File "/usr/local/lib/python3.10/site-packages/anyio/_backends/_asyncio.py", line 914, in run
    result = context.run(func, *args)
  File "/usr/local/lib/python3.10/site-packages/gradio/utils.py", line 826, in wrapper
    response = f(*args, **kwargs)
  File "/home/user/app/app.py", line 956, in <lambda>
    lambda *args: convert_ebook_to_audio(
  File "/home/user/app/app.py", line 801, in convert_ebook_to_audio
    convert_chapters_to_audio_standard_model(chapters_directory, output_audio_directory, temperature, length_penalty, repetition_penalty, top_k, top_p, speed, enable_text_splitting, target_voice, language)
  File "/home/user/app/app.py", line 704, in convert_chapters_to_audio_standard_model
    tts.tts_to_file(
  File "/usr/local/lib/python3.10/site-packages/TTS/api.py", line 432, in tts_to_file
    wav = self.tts(
  File "/usr/local/lib/python3.10/site-packages/TTS/api.py", line 364, in tts
    wav = self.synthesizer.tts(
  File "/usr/local/lib/python3.10/site-packages/TTS/utils/synthesizer.py", line 501, in tts
    print(f" > Real-time factor: {process_time / audio_time}")
ZeroDivisionError: float division by zero
```

**Comment by ROBERT-MCDOWELL** [comment]
it can occur when the machine is very busy or with slow gpu/cpu and a multiple python loop is running for a long time...
what OS and machine you have?

**Comment by geoyogesh** [comment]
Mac

It happens while running with out docker as well 

i added simple try catch arround the tts.tts_to_file( to silence it 

PDF document i used - https://docs.aws.amazon.com/pdfs/wellarchitected/latest/reducing-scope-of-impact-with-cell-based-architecture/reducing-scope-of-impact-with-cell-based-architecture.pdf#what-is-a-cell-based-architecture

**Comment by DrewThomasson** [comment]
## Nope, Its the book he's giving its just super weird cause its a pdf.

### I tested it on a gpu that works for me  And got the Same Error

```python
  File "/home/user/app/app.py", line 704, in convert_chapters_to_audio_standard_model
    tts.tts_to_file(
  File "/home/user/.pyenv/versions/3.10.15/lib/python3.10/site-packages/TTS/api.py", line 334, in tts_to_file
    wav = self.tts(
  File "/home/user/.pyenv/versions/3.10.15/lib/python3.10/site-packages/TTS/api.py", line 276, in tts
    wav = self.synthesizer.tts(
  File "/home/user/.pyenv/versions/3.10.15/lib/python3.10/site-packages/TTS/utils/synthesizer.py", line 507, in tts
    logger.info("Real-time factor: %.3f", process_time / audio_time)
ZeroDivisionError: float division by zero
```

**Comment by DrewThomasson** [comment]
@geoyogesh 

Try giving it a non-pdf document like a `.epub` or a `.txt`

PDF documents are just... super weird lol there is no standard formatting and everyone makes them differently, super hard to extract text from correctly.

I'll see about making it a fix tho

**Comment by DrewThomasson** [comment]
@geoyogesh 
Good to hear the try Catch to silenced it worked on your end tho, tho 👍 :)

### For if any contributers look back at this error this is basically what he was getting in his logs ⬇️ 
[log.txt](https://github.com/user-attachments/files/17806168/log.txt)


> Mac
> 
> It happens while running with out docker as well
> 
> i added simple try catch arround the tts.tts_to_file( to silence it
> 
> PDF document i used - https://docs.aws.amazon.com/pdfs/wellarchitected/latest/reducing-scope-of-impact-with-cell-based-architecture/reducing-scope-of-impact-with-cell-based-architecture.pdf#what-is-a-cell-based-architecture

**Comment by ROBERT-MCDOWELL** [comment]
@geoyogesh I could reproduce it when indeed the sample voice audio file for cloning is missing...
did you try to uploade a custom voice?

**Comment by geoyogesh** [comment]
did you try to uploade a custom voice? -> no...  will that fix it ?

Can you share the headerless command?

**Comment by ROBERT-MCDOWELL** [comment]
no, but your issue will be fixed in version 2.0.0, there is a spacy version conflict with other libraries.

**Comment by DrewThomasson** [comment]
Version 2.0 has just been released which should fix all your issues 

If you issue persists you can reopen this issue :)
